Source/Core/Core/HW/WiimoteEmu/Extension/Classic.h

```diff
@@ -88,47 +88,43 @@ class Classic : public EncryptedExtension
 
   ControllerEmu::ControlGroup* GetGroup(ClassicGroup group);
 
-  enum
-  {
-    PAD_RIGHT = 0x80,
-    PAD_DOWN = 0x40,
-    TRIGGER_L = 0x20,
-    BUTTON_MINUS = 0x10,
-    BUTTON_HOME = 0x08,
-    BUTTON_PLUS = 0x04,
-    TRIGGER_R = 0x02,
-    NOTHING = 0x01,
-    BUTTON_ZL = 0x8000,
-    BUTTON_B = 0x4000,
-    BUTTON_Y = 0x2000,
-    BUTTON_A = 0x1000,
-    BUTTON_X = 0x0800,
-    BUTTON_ZR = 0x0400,
-    PAD_LEFT = 0x0200,
-    PAD_UP = 0x0100,
-  };
-
-  enum
-  {
-    CAL_STICK_CENTER = 0x80,
-    CAL_STICK_RANGE = 0x7f,
-    CAL_STICK_BITS = 8,
-
-    LEFT_STICK_BITS = 6,
-    LEFT_STICK_CENTER_X = CAL_STICK_CENTER >> (CAL_STICK_BITS - LEFT_STICK_BITS),
-    LEFT_STICK_CENTER_Y = CAL_STICK_CENTER >> (CAL_STICK_BITS - LEFT_STICK_BITS),
-    LEFT_STICK_RADIUS = CAL_STICK_RANGE >> (CAL_STICK_BITS - LEFT_STICK_BITS),
-
-    RIGHT_STICK_BITS = 5,
-    RIGHT_STICK_CENTER_X = CAL_STICK_CENTER >> (CAL_STICK_BITS - RIGHT_STICK_BITS),
-    RIGHT_STICK_CENTER_Y = CAL_STICK_CENTER >> (CAL_STICK_BITS - RIGHT_STICK_BITS),
-    RIGHT_STICK_RADIUS = CAL_STICK_RANGE >> (CAL_STICK_BITS - RIGHT_STICK_BITS),
-
-    LEFT_TRIGGER_RANGE = 0x1F,
-    RIGHT_TRIGGER_RANGE = 0x1F,
-  };
-
-  static const u8 STICK_GATE_RADIUS = 0x16;
+  static constexpr u16 PAD_RIGHT = 0x80;
+  static constexpr u16 PAD_DOWN = 0x40;
+  static constexpr u16 TRIGGER_L = 0x20;
+  static constexpr u16 BUTTON_MINUS = 0x10;
+  static constexpr u16 BUTTON_HOME = 0x08;
+  static constexpr u16 BUTTON_PLUS = 0x04;
+  static constexpr u16 TRIGGER_R = 0x02;
+  static constexpr u16 NOTHING = 0x01;
+  static constexpr u16 BUTTON_ZL = 0x8000;
+  static constexpr u16 BUTTON_B = 0x4000;
+  static constexpr u16 BUTTON_Y = 0x2000;
+  static constexpr u16 BUTTON_A = 0x1000;
+  static constexpr u16 BUTTON_X = 0x0800;
+  static constexpr u16 BUTTON_ZR = 0x0400;
+  static constexpr u16 PAD_LEFT = 0x0200;
+  static constexpr u16 PAD_UP = 0x0100;
+
+  static constexpr u8 CAL_STICK_CENTER = 0x80;
+  static constexpr u8 CAL_STICK_RANGE = 0x7f;
+  static constexpr int CAL_STICK_BITS = 8;
```

Comment:
Any particular reason you left those as (signed) int insted of `u32` for example?
Everything else is unsigned, and I can hardly think of a case where _a number of bits_ can be negative.

Reply:
Just for consistency with existing constants in Turntable.cpp. Should I change it? To `u8`?

Reply:
Probably doesn't change too much there; was mostly a question. What does everyone else think about this?

Reply:
I think int is fine here. It's consistent with other constants as JosJuice said and `std::numeric_limits<T>::digits` also uses int for numbers of bits.

I'd pick size_t if we do decide to change them all in the future, though.